composables/zksync/deposit/useTransaction.ts

```diff
@@ -1,23 +1,131 @@
 
 import type { DepositFeeValues } from "@/composables/zksync/deposit/useFee";
 import type { BigNumberish } from "ethers";
-import type { L1Signer } from "zksync-ethers";
 
 export default (getL1Signer: () => Promise<L1Signer | undefined>) => {
   const status = ref<"not-started" | "processing" | "waiting-for-signature" | "done">("not-started");
   const error = ref<Error | undefined>();
   const ethTransactionHash = ref<string | undefined>();
   const eraWalletStore = useZkSyncWalletStore();
+  const { eraNetwork } = useZkSyncProviderStore();
   const { captureException } = useSentryLogger();
+  const chain = eraNetwork.key === "mainnet" ? mainnet : sepolia;
+  const { publicClient, walletClient } = useViemClient({ chain });
 
   const { validateAddress } = useScreening();
 
+  const handleCustomBridgeDeposit = async (
+    transaction: {
+      to: string;
+      tokenAddress: string;
+      amount: BigNumberish;
+      bridgeAddress: string;
+      gasPerPubdataByte?: bigint;
+      l2GasLimit?: bigint;
+      refundRecipient?: string;
+    },
+    fee: DepositFeeValues
+  ) => {
+    // Create signer
+    const { ethereum } = window as any;
+    const browserProvider = new BrowserProvider(ethereum);
+    const isMainnet = eraNetwork.key === "mainnet" || eraNetwork.id === 324;
+    const networkType = isMainnet ? types.Network.Mainnet : types.Network.Sepolia;
+    const zksyncProvider = Provider.getDefaultProvider(networkType);
+    const signer = L1Signer.from(await browserProvider.getSigner(), zksyncProvider);
+
+    if (!publicClient || !walletClient || !zksyncProvider || !signer) return;
+
+    const contract = new Contract(transaction.bridgeAddress, L1_BRIDGE_ABI, signer);
+
+    const l2BridgeAddress = await contract.l2Bridge();
+
+    const to = transaction.to;
+
+    const bridgeData = await getERC20DefaultBridgeData(transaction.tokenAddress, signer._providerL1());
+
+    const wallet = await getL1Signer();
+
+    const l2GasLimit = await signer.providerL2.estimateCustomBridgeDepositL2Gas(
+      transaction.bridgeAddress,
+      l2BridgeAddress,
+      transaction.tokenAddress,
+      transaction.amount.toString(),
+      to!,
+      bridgeData,
+      wallet!.address,
+      transaction?.gasPerPubdataByte ?? 800n,
+      0n
+    );
+
+    const baseCost = await signer.getBaseCost({
+      gasLimit: l2GasLimit,
+      gasPerPubdataByte: transaction.gasPerPubdataByte ?? 800n,
+    });
+
+    const overrides = {
+      gasPrice: fee.gasPrice,
+      gasLimit: fee.l1GasLimit,
+      maxFeePerGas: fee.maxFeePerGas,
+      maxPriorityFeePerGas: fee.maxPriorityFeePerGas,
+    };
+    if (overrides.gasPrice && overrides.maxFeePerGas) {
+      overrides.gasPrice = undefined;
+    }
+
+    const allowance = await signer.getAllowanceL1(transaction.tokenAddress, transaction.bridgeAddress);
+
+    if (allowance < BigInt(transaction.amount)) {
+      const approveTx = await signer.approveERC20(transaction.tokenAddress, transaction.amount, {
```

Comment:
Allowance should be done properly and handled on the UI, there is already logic for it